### PR TITLE
issue205 fix

### PR DESCRIPTION
**Pull request by cnpetra**
fixes a small issue in matrix class when having 0 local vars in the mpi rank

fixes Issue #205

### CODE REVIEW

File: src/LinAlg/hiopMatrixDenseRowMajor.cpp [hunk 1]
```diff
@@ -371,15 +371,13 @@ void hiopMatrixDenseRowMajor::print(FILE* f,
...
-
 /*  y = beta * y + alpha * this * x  
  *
  * Sizes: y is m_local_, x is n_local_, the matrix is m_local_ x n_global_, and the
  * local chunk is m_local_ x n_local_ 
 */
-void hiopMatrixDenseRowMajor::timesVec(double beta, hiopVector& y_,
-			       double alpha, const hiopVector& x_) const
+void hiopMatrixDenseRowMajor::
+timesVec(double beta, hiopVector& y_, double alpha, const hiopVector& x_) const
```
Comment (nychiang) [inline src/LinAlg/hiopMatrixDenseRowMajor.cpp:380]: Each argument on a separate line, i.e., using style guide.
Same comment applied to the other functions in this file.

File: src/Optimization/hiopResidual.cpp [hunk 2]
```diff
@@ -58,11 +58,13 @@ hiopResidual::hiopResidual(hiopNlpFormulation* nlp_)
...
   rd = nlp->alloc_dual_ineq_vec();
-  rxl = rx->alloc_clone();
-  rxu = rx->alloc_clone();
-  rdl = rd->alloc_clone();
-  rdu = rd->alloc_clone();
+  rd->setToZero();
+  rxl = rx->new_copy();
+  rxu = rx->new_copy();
+  rdl = rd->new_copy();
+  rdu = rd->new_copy();
```
Comment (nychiang) [inline src/Optimization/hiopResidual.cpp:67]: I also fixed this in my fr branch. I think it is better to PR it here and then I will rebase my fr to the develop branch.

**Review by nychiang — APPROVED** [review]

File: src/LinAlg/hiopMatrixDenseRowMajor.cpp (hunk 1 above)
Comment (cnpetra) [reply src/LinAlg/hiopMatrixDenseRowMajor.cpp:380]: this is actually valid, arguments can stay on the same line as long as the line is not too long... styleguide needs maybe to be updated to make this more explicit

File: src/Optimization/hiopResidual.cpp (hunk 2 above)
Comment (cnpetra) [reply src/Optimization/hiopResidual.cpp:67]: ok, will merge

**Pull request merged**